boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,932 @@
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ReadRegister(
+    uint8_t   register_address,
+    uint16_t *register_value);
+
+/**
+ * Write data to a specific Serial Input register on the Efuse
+ * @param register_address Serial Input register being written to
+ * @param register_value Value being written to the Serial Input register
+ * @return EXIT_CODE_OK if the write was successful
+ *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ */
+static ExitCode Io_Efuse_WriteRegister(
+    uint8_t         register_address,
+    uint16_t        register_value,
+    GPIO_TypeDef *  chip_select_port,
+    uint16_t        chip_select_pin,
+    struct EfuseIo *e_fuse);
+
+/**
+ * Read data from a specific Serial Output register on the Efuse
+ * @param register_address Serial Output register being read from
+ * @param register_value register contents
+ * @param e_fuse Pointer to Efuse structure being read from
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_ReadRegister(
+    uint8_t         register_address,
+    uint16_t *      register_value,
+    GPIO_TypeDef *  chip_select_port,
+    uint16_t        chip_select_pin,
+    struct EfuseIo *e_fuse);
+
+/**
+ * Calculate the parity of the SPI command to be sent to the Efuse and
+ * set/clear the parity bit. Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
+ * @param spi_command Original SPI command without the parity bit set/cleared
+ */
+static void Io_Efuse_SetParityBit(uint16_t *spi_command);
+
+/**
+ * Write SPI data to Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the write was successful
+ *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ */
+static ExitCode Io_Efuse_WriteToEfuse(
+    uint16_t *    tx_data,
+    GPIO_TypeDef *chip_select_port,
+    uint16_t      chip_select_pin);
+
+/**
+ * Read SPI data from Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param rx_data Pointer to data being received from the Efuse
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *    tx_data,
+    uint16_t *    rx_data,
+    GPIO_TypeDef *chip_select_port,
+    uint16_t      chip_select_pin);
+
+// The structure for each efuse which contains the watchdog-in bit's current
+// value
+static struct EfuseIo aux1_aux2_efuse;
+
+// The SPI handle for the SPI device the E-Fuses are connected to
+static SPI_HandleTypeDef *efuse_spi_handle;
```

Comment:
Can we add the efuse spi handle inside the EfuseIo struct

Reply:
I think that will make things more messy, the spi handle is the same for all of the efuses. I don't think each individual efuse struct needs that as a member

Reply:
I remember doing something similar for the flow meter but Yoshi man suggested this:
https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/564#discussion_r408460904

Reply:
DONE. I was wrong change didn't make things messy lol. Thank you YOSHI MAN 2.0